File: core/JumpGame/JumpGame.java
package com.dbc.code;

public class JumpGame {
    public boolean canJump(int[] nums) {
        int left = nums.length - 1, right = nums.length - 1;
        while (left >= 0) {
            if (nums[left] + left >= right) right = left;
            left--;
        }
        return right == 0;
    }
}

File: core/JumpGame/JumpGame.py
from typing import List


def canJump(nums: List[int]) -> bool:
    left, right = len(nums) - 1, len(nums) - 1
    while left >= 0:
        if nums[left] + left >= right:
            right = left
        left -= 1

    return right == 0


